include/singa/worker.h

```diff
@@ -263,14 +269,26 @@ class Worker {
    */
   inline int grp_id() const { return grp_id_; }
   /**
-   * @reutrn worker ID within the worker group.
+   * @return worker ID within the worker group.
    */
   inline int id() const { return id_; }
 
+  /**
+   * @return The device ID of this worker. It is -1 for CPU and >-1 for GPU.
+   */
+  inline int device_id() { return device_id_; }
```

Comment:
device_id can be accessed via Context. 
it would be better to avoid storing the same variable in two places, which would lead to inconsistent if only one place is updated.